### PR TITLE
fixed hash syntax to 1.9

### PR DESCRIPTION
**Pull request by askl56**

### CODE REVIEW

**Comment by eregon** [comment]
That's a large change!
By curiosity, how did you do it?

**Comment by askl56** [comment]
Using a perl script of all things :P : 

``` find . -name \*.rb -exec perl -p -i -e 's/([^:]):(\w+)\s*=>/\1\2:/g' {} \; ```

**Comment by eregon** [comment]
Nice, thank you for unifying the hash syntax! Style consistency is nice, merged! :+1:

File: core/string/sub_spec.rb [hunk 1]
```diff
@@ -396,7 +396,7 @@
   end
 
   it "ignores non-String keys" do
-    "hello".sub(/(ll)/, 'll' => 'r', :ll => 'z').should == "hero"
+    "hello".sub(/(ll)/, 'll' => 'r', ll: 'z').should == "hero"
```
Comment (kotp) [inline core/string/sub_spec.rb]: Numbers as symbols should be invalid
```
>> "hello".sub(/(11)/, '11' => 'r', 11: 'z')
SyntaxError: (irb):1: syntax error, unexpected ':', expecting =>
"hello".sub(/(11)/, '11' => 'r', 11: 'z')
```

The change here should have been:
```
>> "hello".sub(/(11)/, '11': 'r', 11 => 'z')
"hello"
```
With the quoted number having the syntax change, not the integer.

Comment (askl56) [inline core/string/sub_spec.rb]: Hi @kotp need to catch up in 21-day-challenge gitter soon.

Comment (eregon) [inline core/string/sub_spec.rb]: @kotp It's two times the letter L, downcase, not eleven :wink:.
The CI would definitely have complained if there was such an error.

Comment (kotp) [inline core/string/sub_spec.rb]: The "non-string' keys in the test description and the very similar looking ell's encouraged this.  Perhaps the string itself should be changed for clarity?

Comment (eregon) [inline core/string/sub_spec.rb]: Yes, indeed it could better, but all three have to change. Could you do a PR?

**Comment by kotp** [comment]
See pr #106

**Comment by vais** [comment]
@brauliobo the 1.9 hash syntax is the new hash syntax that was first introduced in 1.9, so it's commonly referred to as the "1.9 syntax". I can see how this could be confusing without the historical context :smile:

**Comment by brauliobo** [comment]
@vais, oh sorry, I must be crazy, I've read the diff the opposite way

**Comment by kotp** [comment]
Numbers as symbols could be another test though, come to think of it.

**Pull request merged**